variation-commons-mongodb/src/main/java/uk/ac/ebi/eva/commons/mongodb/writers/VariantMongoWriter.java

```diff
@@ -148,26 +150,19 @@ private DBObject generateUpdate(Variant variant) {
         if (!addToSet.isEmpty()) {
             update.put("$addToSet", addToSet);
         }
-        update.append("$setOnInsert", convert(variant));
+        update.append("$setOnInsert", convertVariant(variant));
 
         return update;
     }
 
-    private BasicDBList convertStatistics(VariantSourceEntry variantSourceEntry) {
-        List<VariantStatisticsMongo> variantStats = new ArrayList<>();
-        for (Map.Entry<String, VariantStatistics> variantStatsEntry : variantSourceEntry.getCohortStats().entrySet()) {
-            variantStats.add(new VariantStatisticsMongo(
-                    variantSourceEntry.getStudyId(),
-                    variantSourceEntry.getFileId(),
-                    variantStatsEntry.getKey(),
-                    variantStatsEntry.getValue()
-            ));
-        }
-        return (BasicDBList) mongoOperations.getConverter().convertToMongoType(variantStats);
+    private VariantSourceEntry getVariantSourceEntry(Variant variant) {
+        Assert.isTrue(1 == variant.getSourceEntries().size(), "VariantMongoWriter assumes that there's only " +
+                "one study being loaded, so there should only be 0 or 1 VariantSourceEntries inside any Variant");
```

Comment:
Can a variant have 0 VariantSourceEntries? According to the comment, it can. But then the assertion will fail, and the execution will be stopped.

Reply:
Been pointed offline that this method is skipped when there are no Source entries, so the assert won't fail.

Reply:
it's checked outside. this function is only called if `sourceEntries` is not empty